app/jobs/import_versions_job.rb

```diff
@@ -9,8 +9,13 @@ def perform(import)
 
     begin
       import_raw_data(@import.load_data)
-    rescue StandardError => _
-      @import.processing_errors << 'Unknown error occurred'
+    rescue StandardError => error
+      @import.processing_errors << if Rails.env.development?
+                                     "Row #{row}: #{error.message}"
+                                   else
+                                     "Row #{row}: Unknown error occurred"
+                                   end
+      Rails.logger.error "Row #{row}: #{error.message}"
```

Comment:
Uh, there’s no row in this context to log anything about. This only gets errors generated by `each_json_line` (which splits up the import data into JSON objects). Errors that happen processing a given line/row are already handled in `import_raw_data`.

So I think:

1. The data you’re stuffing in `@import.processing_errors` here should be just either `error.message` or `Unknown error occurred`.
2. Likewise the `logger` call here needs to just be `error.message`
3. Both those logger calls should probably include the import ID and/or the import’s file URL, otherwise it’ll be nearly impossible to match up the bad data to the problem.

Reply:
Actually, to this end, we could probably rescue `JSON::ParserError` here (emanating from line 145) and say something more meaningful about it than “unknown error.”